### PR TITLE
GitHub actions: add gardening action

### LINKED ISSUE #114: Actions: add Gardening action
It would be nice to have a gardening that would handle some of the cleanup work we must do on PRs. 

Something like https://github.com/Automattic/action-repo-gardening should do the trick.

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #114 

#### Changes proposed in this Pull Request:

This will allow us to do a few things automatically:

- Add assignee for issues which are being worked on, and adds the "In Progress" label.
- Remove Status labels once a PR has been merged.
- Send a Slack Notification to the Design team to request feedback, based on labels applied to a PR.

For more info:
- https://github.com/Automattic/jetpack/pull/19239
- https://github.com/Automattic/action-repo-gardening

#### Product discussion

Internal reference: p3btAN-1nR-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Check actions running on this repo: https://github.com/Automattic/vaultpress/actions
* Check the labels added to #114. 
* See the Design Input Requested label that was added to this PR after I added the Needs Design Review label, and see the matching message: p1617185707038000-slack-C0D96691V

### CODE REVIEW

File: .github/workflows/gardening.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+       with:
+         node-version: 12
+
+     - name: Wait for prior instances of the workflow to finish
+       uses: softprops/turnstyle@v1
+       env:
+         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+     - name: 'Run gardening action'
+       uses: automattic/action-repo-gardening@master
```
Comment (kraftbj) [inline .github/workflows/gardening.yml:29]: Once we're happy this works, we should tag a version in the action repo and update this. We ran into trouble with the review-required repo that using master elsewhere kinda tied our hands on if we could introduce any major changes (e.g. new config value needed)

**Review by kraftbj — APPROVED** [review]
This all looks good. Approving with the idea that we'd update the version in a follow-up after testing since this is the first implementation outside the JP monorepo.

File: .github/workflows/gardening.yml (hunk 1 above)
Comment (jeherve) [reply .github/workflows/gardening.yml:29]: Yeah, that sound good to me. 👍

**Comment by jeherve** [comment]
Label gets cleaned up properly. This confirms the last task (`cleanLabels`) works too.

File: .github/workflows/gardening.yml (hunk 1 above)
Comment (jeherve) [reply .github/workflows/gardening.yml:29]: Here is the update: #116

**Pull request merged**